molgenis-api-permissions/src/main/java/org/molgenis/api/permissions/model/response/PermissionResponse.java

```diff
@@ -0,0 +1,33 @@
+package org.molgenis.api.permissions.model.response;
+
+import com.google.auto.value.AutoValue;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.molgenis.util.AutoGson;
+
+@AutoValue
+@AutoGson(autoValueClass = AutoValue_PermissionResponse.class)
+@SuppressWarnings(
+    "squid:S1610") // Abstract classes without fields should be converted to interfaces
+public abstract class PermissionResponse {
+
+  @Nullable
+  public abstract String getUser();
+
+  @Nullable
+  public abstract String getRole();
+
+  @Nullable
+  public abstract String getPermission();
+
+  @Nullable
+  public abstract Set<LabelledPermissionResponse> getInheritedPermissions();;
```

Comment:
Should return an empty set instead of null?

Reply:
It does, when inherited permissions are requested, null if this is not the case.